modules/unsupported/wfs-ng/src/test/java/org/geotools/data/wfs/internal/WFSRequestTest.java

```diff
@@ -0,0 +1,110 @@
+package org.geotools.data.wfs.internal;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.logging.Level;
+import org.geotools.data.ows.HTTPClient;
+import org.geotools.data.ows.SimpleHttpClient;
+import org.geotools.data.wfs.WFSTestData;
+import org.geotools.data.wfs.internal.WFSConfig.PreferredHttpMethod;
+import org.geotools.ows.ServiceException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/*
+ * GeoTools - The Open Source Java GIS Toolkit http://geotools.org
+ *
+ * (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ * This library is free software; you can redistribute it and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/** @author Matthias Schulze - Landesamt für Digitalisierung, Breitband und Vermessung */
+public class WFSRequestTest {
+
+    @Before
+    public void setUp() throws Exception {
+        Loggers.RESPONSES.setLevel(Level.FINEST);
+    }
+
+    /** Test method for {@link org.geotools.data.wfs.internal.WFSRequest#getFinalURL()}. */
+    @Test
+    public void testGetFinalURLPost() throws IOException, ServiceException {

```

Comment:
This is checking the query string, but I don't see any test for the filter capabilties changes? There needs to be some tests for the capabilities detection.

Reply:
Which tests cases would you suggest?

Reply:
I'm not much familiar with this module (well nobody is, or it would be supported I guess), but I see ``AbstractWFSStrategy.splitFilters`` is public, so can you make a store parse the capabiliites, get the strategy, make it split filters, and check that the filters are in the pre-filter as opposed to the post-filter?
There is something like that going on here: https://github.com/geotools/geotools/blob/232332f08b8e02a6dcc62b0d5001230981bdaa8b/modules/unsupported/wfs-ng/src/test/java/org/geotools/data/wfs/integration/IntegrationTestWFSClient.java#L220